app/assets/javascripts/spotlight/reindex_monitor.js

```diff
@@ -0,0 +1,109 @@
+Spotlight.onLoad(function() {
+  $('[data-behavior="reindex-monitor"]').reindexMonitor();
+});
+
+(function($) {
+  $.fn.reindexMonitor = function() {
+    var monitorElements = this;
+    var defaultRefreshRate = 3000;
+    var panelContainer;
+
+    $(monitorElements).each(function() {
+      panelContainer = $(this);
+      var monitorUrl = panelContainer.data('monitorUrl');
+      var refreshRate = panelContainer.data('refreshRate') || defaultRefreshRate;
+      setInterval(function() {
+        checkMonitorUrl(monitorUrl);
+      }, refreshRate);
+    });
+
+    function checkMonitorUrl(url) {
+      $.ajax(url).success(success).fail(fail);
+    }
+
+    function success(data) {
+      if (data.in_progress) {
```

Comment:
This is coming from a ruby method and I tend to prefer to keep to ruby's style, but could be convinced to change the keys in the JSON we return (or perhaps relax the jshint rule).